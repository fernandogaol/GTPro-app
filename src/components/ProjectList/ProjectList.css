.ProjectList {
  max-height: 600px;
  display: flex;
  flex-direction: row;
}
