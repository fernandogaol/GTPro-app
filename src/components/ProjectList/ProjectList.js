import React, { Component } from 'react';
import ProjectApiService from '../../services/projects-api-service';
import DashboardProjects from '../DashboardProjects/DashboardProjects';
import UserContext from '../../context/ProjectsForUserContext';
import './ProjectList.css';
// import UserContext from '../../context/UserContext';

export default class ProjectList extends Component {
  // static contextType = ProjectListContext;
  static contextType = UserContext;

  componentDidMount(res) {
    this.context.clearError();
    ProjectApiService.getProject(this.context.setProjectId);
    ProjectApiService.getProject(this.context.userId)
      .then(this.context.setProject)
      .catch(this.context.setError);
  }
  renderProjects() {
    const { project = [] } = this.context;
    return project.map((project) => (
      <DashboardProjects project={project} key={project.id} />
    ));
  }
  handleSubmit = (ev) => {
    ev.preventDefault();
    const { title } = ev.target;

    this.setState({ error: null });

    ProjectApiService.postProject({
      title: title.value,
    })
      .then((newProject) => {
        title.value = '';
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error } = this.context;
    return (
      <section className='ProjectList'>
        {error ? (
          <p className='red'> There was an error, please try again</p>
        ) : (
          this.renderProjects()
        )}
        <form onSubmit={this.handleSubmit}>
          <textarea name='title' type='text' requinput></textarea>
          <button>Add project +</button>
        </form>
      </section>
    );
  }
}
