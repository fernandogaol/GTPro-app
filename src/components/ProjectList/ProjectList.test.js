import React from 'react';
import renderer from 'react-test-renderer';
import ProjectList from './ProjectList';

it('renders correctly', () => {
  const tree = renderer.create(<ProjectList />).toJSON();
  expect(tree).toMatchSnapshot();
});
