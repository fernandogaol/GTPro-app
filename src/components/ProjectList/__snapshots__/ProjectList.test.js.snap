// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="ProjectList"
>
  <h1>
    Board
  </h1>
  <div
    className="renderProjects"
  />
  <form
    className="AddProjectForm"
    onSubmit={[Function]}
  >
    <input
      id="titleInput"
      name="title"
      placeholder="Add new project"
      type="text"
    />
    <button
      id="addBtn"
      type="submit"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        id="plusBtn"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </form>
</section>
`;
